### DISCUSSION TITLE
XFS volume RW mount issue

### DISCUSSION THREAD

**Discussion by chernega89** [longhorn/longhorn#3472] [Q&A]
Hello, 

I have an issue with mounting XFS volume (using Longhorn-UI or with manually created storage class and PVC) in RW to pods on my setup (K8S - v1.22.4 , Longhorn - 1.2.3).  There are no issues with EXT4 volumes and with mounting XFS in RO.  Maybe someone could advice. 

Logs: 

  Type     Reason                  Age   From                     Message
  ----     ------                  ----  ----                     -------
  Normal   Scheduled               17s   default-scheduler        Successfully assigned default/pod-pvc3 to k8s-mws-cr-w03
  Normal   SuccessfulAttachVolume  6s    attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-31f19966-8dfa-4a0f-af04-bbae361e95db"
  Warning  FailedMount             0s    kubelet                  MountVolume.MountDevice failed for volume "pvc-31f19966-8dfa-4a0f-af04-bbae361e95db" : rpc error: code = Internal desc = mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t xfs -o defaults /dev/longhorn/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db/globalmount
Output: mount: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db/globalmount: wrong fs type, bad option, bad superblock on /dev/longhorn/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db, missing codepage or helper program, or other error.

**[07:10:25]  kubectl get storageclasses.storage.k8s.io longhorn-xfs -o yaml**
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2021-12-28T12:04:21Z"
  name: longhorn-xfs
  resourceVersion: "4298444"
  uid: 7e0c3809-d5d4-4fef-8462-db61e7213da6
parameters:
  fromBackup: ""
  fsType: xfs
  numberOfReplicas: "2"
  staleReplicaTimeout: "2880"
provisioner: driver.longhorn.io
reclaimPolicy: Delete
volumeBindingMode: Immediate

**[07:11:49]  kubectl get pvc**
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
longhorn-volv-pvc   Bound    pvc-31f19966-8dfa-4a0f-af04-bbae361e95db   2Gi        RWO            longhorn-xfs   6m19s

**[07:11:52] kubectl get pv**
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                       STORAGECLASS   REASON   AGE
pvc-31f19966-8dfa-4a0f-af04-bbae361e95db   2Gi        RWO            Delete           Bound    default/longhorn-volv-pvc   longhorn-xfs            6m20s

**[07:11:55] kubectl describe pv pvc-31f19966-8dfa-4a0f-af04-bbae361e95db** 
Name:            pvc-31f19966-8dfa-4a0f-af04-bbae361e95db
Labels:          <none>
Annotations:     pv.kubernetes.io/provisioned-by: driver.longhorn.io
Finalizers:      [kubernetes.io/pv-protection external-attacher/driver-longhorn-io]
StorageClass:    longhorn-xfs
Status:          Bound
Claim:           default/longhorn-volv-pvc
Reclaim Policy:  Delete
Access Modes:    RWO
VolumeMode:      Filesystem
Capacity:        2Gi
Node Affinity:   <none>
Message:         
Source:
    Type:              CSI (a Container Storage Interface (CSI) volume source)
    Driver:            driver.longhorn.io
    FSType:            xfs
    VolumeHandle:      pvc-31f19966-8dfa-4a0f-af04-bbae361e95db
    ReadOnly:          false
    VolumeAttributes:      fromBackup=
                           fsType=xfs
                           numberOfReplicas=2
                           staleReplicaTimeout=2880
                           storage.kubernetes.io/csiProvisionerIdentity=1640690222192-8081-driver.longhorn.io
Events:                <none>

**[07:12:50] kubectl describe pvc longhorn-volv-pvc** 
Name:          longhorn-volv-pvc
Namespace:     default
StorageClass:  longhorn-xfs
Status:        Bound
Volume:        pvc-31f19966-8dfa-4a0f-af04-bbae361e95db
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: driver.longhorn.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      2Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Used By:       pod-pvc3
Events:
  Type    Reason                 Age                    From                                                                                      Message
  ----    ------                 ----                   ----                                                                                      -------
  Normal  Provisioning           7m33s                  driver.longhorn.io_csi-provisioner-66cdd9875f-9zb4h_f7cae247-886e-4c1d-8584-4d54b6bdba8c  External provisioner is provisioning volume for claim "default/longhorn-volv-pvc"
  Normal  ExternalProvisioning   7m33s (x2 over 7m33s)  persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "driver.longhorn.io" or manually created by system administrator
  Normal  ProvisioningSucceeded  7m31s                  driver.longhorn.io_csi-provisioner-66cdd9875f-9zb4h_f7cae247-886e-4c1d-8584-4d54b6bdba8c  Successfully provisioned volume pvc-31f19966-8dfa-4a0f-af04-bbae361e95db

**[06:23:33]  dmesg | tail - worker node where the block device is attached and pod is running.** 
[Tue Dec 28 07:12:42 2021] XFS (sdb): Filesystem can only be safely mounted read only.
[Tue Dec 28 07:12:42 2021] XFS (sdb): SB validate failed with error -22.
[Tue Dec 28 07:14:45 2021] XFS (sdb): Superblock has unknown read-only compatible features (0x4) enabled.
[Tue Dec 28 07:14:45 2021] XFS (sdb): Attempted to mount read-only compatible filesystem read-write.
[Tue Dec 28 07:14:45 2021] XFS (sdb): Filesystem can only be safely mounted read only.
[Tue Dec 28 07:14:45 2021] XFS (sdb): SB validate failed with error -22.
[Tue Dec 28 07:16:47 2021] XFS (sdb): Superblock has unknown read-only compatible features (0x4) enabled.
[Tue Dec 28 07:16:47 2021] XFS (sdb): Attempted to mount read-only compatible filesystem read-write.
[Tue Dec 28 07:16:47 2021] XFS (sdb): Filesystem can only be safely mounted read only.
[Tue Dec 28 07:16:47 2021] XFS (sdb): SB validate failed with error -22.

**[07:19:11]  mount -t xfs -o defaults /dev/longhorn/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db /mnt/**
mount: wrong fs type, bad option, bad superblock on /dev/longhorn/pvc-31f19966-8dfa-4a0f-af04-bbae361e95db,
       missing codepage or helper program, or other error

Any assistance will be much appreciated because I am completely confused (((. 

Thank you

**Comment by jenting** [comment]
we had a similar [issue](https://github.com/longhorn/longhorn/issues/2642) for this one before.

@joshimoo, do you have any feedback on this one?
@khushboo-rancher, could you please try to reproduce it on Longhorn v1.2.3 with XFS volume?

Thank you all.

**Reply by joshimoo** [reply]
@chernega89 what is your host os? Is this a restored volume? 
Knowledge sharing: xfs can only mount one volume with UUID XXX

From the above it might be a version mismatch (since your xfs volume was created with a feature set, that your mounter does not support), did you change/update host os?
Was this volume restored on a different cluster?

Some refs:
https://www.reddit.com/r/CentOS/comments/ng2q3v/xfs_from_centos_83_can_only_be_mounted_readonly/
https://github.com/ceph/ceph-csi/issues/966
